report_pre_thesis/src/sections/methodology.tex

```diff
@@ -1,8 +1,246 @@
+\end{figure}
+
+The regression process is illustrated in figure \ref{fig:ica_regression}.
+Since we are implementing this recreation in Python, a different programming language than the original pipeline, we cannot use the same libraries for regression.
+We use the \texttt{LinearRegression} class from the \texttt{scikit-learn} library\cite{scikit-learn}.
+However, \citet{cleggRecalibrationMarsScience2017} mentions performing four more types of regression in addition to linear regression, namely parabolic, exponential, geometric and logsquare.
+For each oxide, they used the type of regression and normalization that produced the best results.
+
+% TODO: Move this table to the results section and embed our own results in it as well to show the baseline results compared to the original results.
+% This can be seen in table \ref{tab:regression_types}, which is taken from \citet{cleggRecalibrationMarsScience2017}.
+
+% \begin{table}[h]
+% \centering
+% \begin{tabular*}{\columnwidth}{@{\extracolsep{\fill}}lccc}
+% \toprule
+
+% Element & Norm & Law type    & RMS \\ \midrule
+% Si      & 1    & Logsqaure   & 6.7 \\
+% Ti      & 3    & Geometric   & 0.6 \\
+% Al      & 3    & Geometric   & 4.4 \\
+% Fe      & 1    & Geometric   & 2.2 \\
+% Mg      & 1    & Exponential & 3.0 \\
+% Ca      & 1    & Parabolic   & 1.0 \\
+% Na      & 3    & Parabolic   & 0.6 \\
+% K       & 3    & Geometric   & 0.4 \\
+% \bottomrule
+% \end{tabular*}
+% \caption{The summary of each ICA model characteristics from \citet{cleggRecalibrationMarsScience2017}.}
+% \label{tab:regression_types}
+% \end{table}
+
+% TODO: Update/rewrite this to reflect the new curve-fit implementation.
+% Because \texttt{scikit-learn} does not support these types of regression, we instead opted to transform the data to fit the linear regression model.
+% For example, to perform exponential regression, we take the natural logarithm of the data, perform linear regression, and then take the exponential of the result.
+% This is done for each of the four types of regression mentioned above.
+% The code for this can be seen in listing \ref{lst:data_transformation}.
+
+% \begin{lstlisting}[
+% 	language=Python,
+% 	caption={The code for transforming the data to fit the linear regression model.},
+% 	label={lst:data_transformation}
+% 	]
+% if model_name == "Log-square":
+% 	X_train = np.log(X_train**2)
+% 	X_test = np.log(X_test**2)
+% elif model_name == "Exponential":
+% 	X_train = np.log(X_train)
+% 	X_test = np.log(X_test)
+% elif model_name == "Geometric":
+% 	X_train = np.sqrt(X_train)
+% 	X_test = np.sqrt(X_test)
+% elif model_name == "Parabolic":
+% 	X_train = np.column_stack((X_train, X_train**2))
+% 	X_test = np.column_stack((X_test, X_test**2))
+% \end{lstlisting}
+
+In summary, we have created a replica of the ICA phase of the pipeline with strategic adaptations to accommodate the differences in programming language and the missing or unclear information about the original implementation.
```

Comment:
Written by Copilot? 

The difference in programming language argument is weak. Please don't use it.
`Strategic adaptations` is mostly OK for not removing the IRF and somewhat OK for outlier removal. Could you rephrase?

Reply:
We removed it because a summary of a short subsection didn't feel right at all - and with the newly added text, it already ends in a pretty nice way, I think.